crates/oxc_linter/src/rules/import/namespace.rs

```diff
@@ -1,30 +1,104 @@
+use oxc_ast::{ast::JSXElementName, AstKind};
 use oxc_diagnostics::{
     miette::{self, Diagnostic},
     thiserror::Error,
 };
 use oxc_macros::declare_oxc_lint;
-use oxc_span::{CompactString, Span};
+use oxc_semantic::AstNode;
+use oxc_span::{GetSpan, Span};
 
 use crate::{context::LintContext, rule::Rule};
 
 #[derive(Debug, Error, Diagnostic)]
-#[error("eslint-plugin-import(namespace): ")]
-#[diagnostic(severity(warning), help(""))]
-struct NamespaceDiagnostic(CompactString, #[label] pub Span);
+enum NamespaceDiagnostic {
+    #[error("eslint-plugin-import(namespace): {1:?} not found in imported namespace {2:?}.")]
+    #[diagnostic(severity(warning))]
+    NoExport(#[label] Span, String, String),
+    #[error("eslint-plugin-import(namespace): Unable to validate computed reference to imported namespace {1:?}.")]
+    #[diagnostic(severity(warning))]
+    ComputedReference(#[label] Span, String),
+    #[error("eslint-plugin-import(namespace): Assignment to member of namespace {1:?}.'")]
+    #[diagnostic(severity(warning))]
+    Assignment(#[label] Span, String),
+}
 
 /// <https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/namespace.md>
 #[derive(Debug, Default, Clone)]
 pub struct Namespace;
 
 declare_oxc_lint!(
     /// ### What it does
-    /// TODO
+    /// Enforces names exist at the time they are dereferenced, when imported as a full namespace (i.e. import * as foo from './foo'; foo.bar(); will report if bar is not exported by ./foo.).
+    /// Will report at the import declaration if there are no exported names found.
+    /// Also, will report for computed references (i.e. foo["bar"]()).
+    /// Reports on assignment to a member of an imported namespace.
     Namespace,
     nursery
 );
 
 impl Rule for Namespace {
-    fn run_once(&self, _ctx: &LintContext<'_>) {}
+    fn run(&self, node: &AstNode<'_>, ctx: &LintContext<'_>) {
+        if let AstKind::ImportNamespaceSpecifier(specifier) = node.kind() {
+            let Some(AstKind::ImportDeclaration(decl)) = ctx.nodes().parent_kind(node.id()) else {
+                return;
+            };
+
+            let source = &decl.source.value;
+            let module_record = ctx.semantic().module_record();
+            let Some(module) = module_record.loaded_modules.get(source.as_ref()) else {
+                return;
+            };
+
+            let check_binding_exported = |name: &str, span| {
+                if module.exported_bindings.get(name).is_none() {
```

Comment:
I added a `ModuleRecord::no_esm` yesterday, we need to skip all commonjs modules.

Reply:
The early skip can be done after the check

```
            let Some(module) = module_record.loaded_modules.get(source.as_ref()) else {
                return;
            };
```